bigquery-connector-common/src/test/java/com/google/cloud/bigquery/connector/common/BigQueryUtilTest.java

```diff
@@ -303,9 +309,48 @@ public void testSchemaWritableWithDisableNullableFieldCheck() {
         Schema.of(
             Field.newBuilder("foo", StandardSQLTypeName.STRING).setMode(Mode.REPEATED).build());
 
+    assertThat(BigQueryUtil.schemaWritable(s1, s1, false, false)).isTrue();
+    assertThat(BigQueryUtil.schemaWritable(s2, s2, false, false)).isTrue();
+    assertThat(BigQueryUtil.schemaWritable(s3, s3, false, false)).isTrue();
     assertThat(BigQueryUtil.schemaWritable(s1, s2, false, false)).isTrue();
     assertThat(BigQueryUtil.schemaWritable(s1, s3, false, false)).isTrue();
     assertThat(BigQueryUtil.schemaWritable(s2, s3, false, false)).isTrue();
+    assertThat(BigQueryUtil.schemaWritable(s2, s1, false, false)).isTrue();
+    assertThat(BigQueryUtil.schemaWritable(s3, s1, false, false)).isTrue();
+    assertThat(BigQueryUtil.schemaWritable(s3, s2, false, false)).isTrue();
+  }
+
+  @Test
+  public void testSchemaWritableWithMoreUnEqualNumberOfFields() {
+    Schema s1 =
+        Schema.of(
+            Field.newBuilder("foo1", StandardSQLTypeName.STRING).setMode(Mode.NULLABLE).build(),
+            Field.newBuilder("foo2", StandardSQLTypeName.STRING).setMode(Mode.NULLABLE).build());
+    Schema s2 =
+        Schema.of(
+            Field.newBuilder("foo1", StandardSQLTypeName.STRING).setMode(Mode.NULLABLE).build(),
+            Field.newBuilder("foo2", StandardSQLTypeName.STRING).setMode(Mode.NULLABLE).build(),
+            Field.newBuilder("foo3", StandardSQLTypeName.STRING).setMode(Mode.NULLABLE).build());
+    Schema s3 =
+        Schema.of(
+            Field.newBuilder("foo1", StandardSQLTypeName.STRING).setMode(Mode.NULLABLE).build());
+
+    assertThat(BigQueryUtil.schemaWritable(s1, s2, false, true)).isTrue();
+    assertThat(BigQueryUtil.schemaWritable(s1, s3, false, true)).isFalse();
+    assertThat(BigQueryUtil.schemaWritable(s3, s2, false, true)).isTrue();
+  }
+
+  @Test
+  public void testIsModeWritable() {
+    assertThat(BigQueryUtil.isModeWritable(Mode.NULLABLE, Mode.NULLABLE)).isTrue();
+    assertThat(BigQueryUtil.isModeWritable(Mode.NULLABLE, Mode.REQUIRED)).isTrue();
```

Comment:
Why writing nullable to required is ok? it may contain `null`s

Reply:
So here we are only checking the schema not the actual value. 
So here is the case if the destination field is `REQUIRED` and the source is `NULLABLE`, but if the source contains non-null value, it should be written. The check here is to see if schema is compatible not the values.